components/EventDetail.js

```diff
@@ -22,30 +32,137 @@ const StyledEventInfoContainer = styled.div`
 const StyledEventInfo = styled.p`
   justify-self: ${(props) => (props.$right ? "end" : "start")};
 `;
-const StyledDescription = styled.p``;
 
-export default function EventDetail({ event }) {
-  const { day, month, year, time } = getDate(event.startDateTime);
+const StyledDescription = styled.p`
+  font-size: 1rem;
+  grid-column: 1 / -1;
+`;
+
+// form styles
+const StyledForm = styled.form`
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  justify-content: space-between;
+  align-items: baseline;
+`;
+
+const StyledLabel = styled.label`
+  font-size: 1rem;
+  display: flex;
+  flex-direction: column;
+  grid-column: ${(props) => (props.$full ? "1 / -1" : "")};
+`;
+
+const StyledInput = styled.input`
+  font-size: 1rem;
+  border: 1px solid #000000;
+  border-radius: 5px;
+`;
+
+const StyledTextarea = styled.textarea`
+  font-size: 1rem;
+  border: 1px solid #000000;
+  border-radius: 5px;
+`;
+
+export default function EventDetail({ event = {} }) {
+  const { mutate } = useSWR(`/api/events/${event._id}`);
+  // destructuring a formatted date of the event object
+  const { day, month, year, formattedDate, time } = getDate(
+    event.startDateTime
+  );
+
+  // toggle for Edit Mode
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  // handle functions
+  async function handleSubmit(jsEvent) {
```

Comment:
we usually call this `event` here in class , you called it event at other places in your code. Try to stick to one naming. If possible use the conventional one.